internal/resources/sample_bucket.go

```diff
@@ -0,0 +1,376 @@
+func (s *SampleBucket) retrieveSampleBucket(ctx context.Context, organizationId, projectId, clusterId, bucketId string) (*providerschema.SampleBucket, error) {
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/sampleBuckets/%s", s.HostURL, organizationId, projectId, clusterId, bucketId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := s.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		s.Token,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrExecutingRequest, err)
+	}
+
+	sampleBucketResp := samplebucketapi.GetSampleBucketResponse{}
+	err = json.Unmarshal(response.Body, &sampleBucketResp)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errors.ErrUnmarshallingResponse, err)
+	}
+	sampleStats := providerschema.NewStats(*sampleBucketResp.Stats)
+	sampleBucketStatsObj, diags := types.ObjectValueFrom(ctx, sampleStats.AttributeTypes(), sampleStats)
+	if diags.HasError() {
+		return nil, errors.ErrUnableToConvertAuditData
+	}
+
+	refreshedState := providerschema.SampleBucket{
+		Id:                       types.StringValue(sampleBucketResp.Id),
+		Name:                     types.StringValue(sampleBucketResp.Name),
+		OrganizationId:           types.StringValue(organizationId),
+		ProjectId:                types.StringValue(projectId),
+		ClusterId:                types.StringValue(clusterId),
+		Type:                     types.StringValue(sampleBucketResp.Type),
+		StorageBackend:           types.StringValue(sampleBucketResp.StorageBackend),
+		MemoryAllocationInMB:     types.Int64Value(sampleBucketResp.MemoryAllocationInMb),
+		BucketConflictResolution: types.StringValue(sampleBucketResp.BucketConflictResolution),
+		DurabilityLevel:          types.StringValue(sampleBucketResp.DurabilityLevel),
+		Replicas:                 types.Int64Value(sampleBucketResp.Replicas),
+		Flush:                    types.BoolValue(sampleBucketResp.Flush),
+		TimeToLiveInSeconds:      types.Int64Value(sampleBucketResp.TimeToLiveInSeconds),
+		EvictionPolicy:           types.StringValue(sampleBucketResp.EvictionPolicy),
+		Stats:                    sampleBucketStatsObj,
+	}
+
+	return &refreshedState, nil
+}
+
+func isValidSampleName(category string) bool {
+	switch category {
+	case
+		"travel-sample",
+		"beer-sample",
+		"gamesim-sample":
+		return true
+	}
+	return false
+}
+
+// initializeBucketWithPlanAndId initializes an instance of providerschema.Bucket
+// with the specified plan and ID. It marks all computed fields as null.
+func initializeSampleBucketWithPlanAndId(plan providerschema.SampleBucket, id string) providerschema.SampleBucket {
+	plan.Id = types.StringValue(id)
+	if plan.StorageBackend.IsNull() || plan.StorageBackend.IsUnknown() {
+		plan.StorageBackend = types.StringNull()
+	}
+	if plan.EvictionPolicy.IsNull() || plan.EvictionPolicy.IsUnknown() {
+		plan.EvictionPolicy = types.StringNull()
+	}
```

Comment:
Since we have changed the optional and computed fields, this function should also be modified